### PR TITLE
LPD-39122 Step tracker listens to event

### PR DESCRIPTION
**Pull request by andrea-ale-sbarra**
https://liferay.atlassian.net/browse/LPD-39122

### CODE REVIEW

File: modules/apps/commerce/commerce-order-content-web/src/main/resources/META-INF/resources/js/StepTracker.js [hunk 1]
```diff
@@ -3,19 +3,61 @@
...
 import React from 'react';
+import {getOrder} from "./util";
 
 const StepTracker = ({stepModels}) => {
+
+	const [cartSteps, setCartSteps] = useState(stepModels);
+
+	const onStatusChange = useCallback(
+		({order = null}) => {
+			getOrder(isOpen, order, orderId)
```
Comment (gianmarcobrunialti) [inline modules/apps/commerce/commerce-order-content-web/src/main/resources/META-INF/resources/js/StepTracker.js:17]: You could skip a network req here.

```javascript
import {getOrder as getOrderFromAPI} from './util';

// ... 

const onStatusChange = useCallback(
  ({order = null}) => {
    const getOrder = order?.cartSteps ? getOrderFromAPI : () => Promise.resolve(order);

    getOrder(isOpen, order, orderId)
      .then((order) => {
        setCartSteps(order.cartSteps);
    })
  
  // ...
```

**Review by gianmarcobrunialti — COMMENTED** [review]
LGTM code-wise, just a functional suggestion.

**Comment by gianmarcobrunialti** [comment]
Discussed offline: expose stepModels for delivery PlacedOrder API as well.

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Review by gianmarcobrunialti — APPROVED** [review]
G2G

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
Forwarded here: https://github.com/liferay-commerce/liferay-portal/pull/5514

**Pull request closed without merging**